admin/api.go

```diff
@@ -165,6 +163,12 @@ func (a *AdminAPI) GetConnectors() ([]connector.ConnectorConfig, error) {
 }
 
 func mapError(e error) error {
+	switch t := e.(type) {
+	case client.ValidationError:
```

Comment:
I really dislike type switching for a specific error since it makes the validateClient code really brittle. Would be nice if this was formalized more, such as introducing HTTP specific [error behavior](http://dave.cheney.net/2014/12/24/inspecting-errors). I suppose that's outside of the scope of this PR though.